gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/Manifest.java

```diff
@@ -0,0 +1,117 @@
+package org.apache.gobblin.data.management.copy;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+/**
+ * Manifest schema and serDe
+ * https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Manifest+based+distcp+runbook
+ */
+public class Manifest {
+  private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+  private static final Type CopyableUnitListType = new TypeToken<ArrayList<CopyableUnit>>(){}.getType();
+
+
+  public final ArrayList<CopyableUnit> _copyableUnits;
+
+  public Manifest() {
+    _copyableUnits = new ArrayList<>();
+  }
+
+  public Manifest(ArrayList<CopyableUnit> copyableUnits) {
+    _copyableUnits = copyableUnits;
+  }
+
+  public void add(Manifest.CopyableUnit copyableUnit) {
+    _copyableUnits.add(copyableUnit);
+  }
+
+  public static class CopyableUnit {
+    public final Integer id;
+    public final String fileName;
+    public final String fileGroup;
```

Comment:
Also do we need to make these fields (fileGroup, fileSize, and modTime) as optional?

Reply:
they are optional by passing null. The constructor needs all params for json parsing. That is why we use Integer instead of int and Long instead of long

Reply:
we could add some extra validation in read/write to ensure `filename` is there.

Reply:
added validation to read iterator path and write path and left a comment on read that validation is not used